packages/react/src/components/TimePicker/TimePicker.jsx

```diff
@@ -0,0 +1,572 @@
+  /** Size of input */
+  size: PropTypes.oneOf(['sm', 'md', 'lg']),
+  /** Specify whether the control is currently disabled */
+  disabled: PropTypes.bool,
+  /** Specify whether the control is currently read only */
+  readOnly: PropTypes.bool,
+  /** Specify whether the control is currently in warning state */
+  warn: PropTypes.arrayOf(PropTypes.bool),
+  /** Specify whether the control is currently in invalid state */
+  invalid: PropTypes.arrayOf(PropTypes.bool),
+  /** Boolean to swap for light theme */
+  light: PropTypes.bool,
+  /** Optional handler that is called whenever <input> is updated - will be called with new value as an argument */
+  onChange: PropTypes.func,
+  testId: PropTypes.string,
+};
+
+/* istanbul ignore next */
+const defaultProps = {
+  className: undefined,
+  hideLabel: false,
+  value: undefined,
+  secondaryValue: undefined,
+  hideSecondaryLabel: false,
+  type: 'single',
+  i18n: {
+    invalidText: 'The time entered is invalid',
+    warnText: undefined,
+    timeIconText: 'Open time picker',
+    placeholderText: 'hh:mm',
+    readOnlyBtnText: 'Read only',
+  },
+  size: 'md',
+  disabled: false,
+  readOnly: false,
+  warn: [false, false],
+  invalid: [false, false],
+  light: false,
+  onChange: () => {},
+  testId: 'time-picker',
+};
+
+const validate = (newValue) => {
+  const isValid12HoursRegex = /^((0[1-9])?|(1[0-2])?)*:[0-5][0-9] (AM|PM)$/;
+  return isValid12HoursRegex.test(newValue) || newValue === '';
+};
+
+const TimePicker = ({
+  id,
+  className,
+  type,
+  i18n,
+  size,
+  disabled,
+  testId,
+  warn,
+  invalid,
+  light,
+  readOnly,
+  hideLabel,
+  hideSecondaryLabel,
+  value,
+  secondaryValue,
+  onChange,
+}) => {
+  const init = useRef(false);
+  const inputRef = useRef();
+  const secondaryInputRef = useRef();
+  const dropDownRef = useRef();
+  const container = inputRef.current;
+  const invalidProp = invalid[0];
+  const secondaryInvalidProp = invalid[1];
+  const warnProp = warn[0];
+  const secondaryWarnProp = warn[1];
+  // Component state
+  const [openState, setOpenState] = useState(false);
+  const [position, setPosition] = useState([0, 0]);
+  const [focusedInput, setFocusedInput] = useState(0);
+  const [valueState, setValueState] = useState(value || '');
+  const [secondaryValueState, setSecondaryValueState] = useState(secondaryValue || '');
+  const [invalidState, setInvalidState] = useState(invalidProp);
+  const [secondaryInvalidState, setSecondaryInvalidState] = useState(secondaryInvalidProp);
+
+  useEffect(() => {
+    if (init.current) {
+      onChange(valueState, secondaryValueState);
+    } else {
+      init.current = true;
+    }
+  }, [valueState, secondaryValueState, onChange, init]);
+
+  useEffect(() => {
+    setInvalidState(invalidProp);
+    setSecondaryInvalidState(secondaryInvalidProp);
+  }, [invalidProp, secondaryInvalidProp]);
+
+  useEffect(() => {
+    if (container) {
+      const { left, bottom } = container.getBoundingClientRect();
```

Comment:
Looking at this now we may want to put some logic here to smartly set position of dropdown. I believe we have a positioning hook to use to this effect.